com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/pipeline/shader/ShaderCompilePipeline.java

```diff
@@ -292,9 +292,17 @@ public byte[] crossCompile(ShaderDesc.ShaderType shaderType, ShaderDesc.Language
         ShaderModule module = getShaderModule(shaderType);
         assert module != null;
 
-        if (shaderLanguage == ShaderDesc.Language.LANGUAGE_SPIRV) {
+        boolean isLanguageGLSL = shaderLanguageIsGLSL(shaderLanguage);
+
+        if (isLanguageGLSL && module.shaderInfo != null && module.shaderInfo.version == version) {
+            // The input shader is already valid GLSL code in the output version we are trying to produce,
+            // no need to crosscompile!
+            return module.desc.source.getBytes();
+        } else if (shaderLanguage == ShaderDesc.Language.LANGUAGE_SPIRV) {
```

Comment:
Now that I think about it, we won't benefit from shader optimizations from the pipeline now since we are sidestepping the SPIR-v pipeline.. Hm, something to think about